src/core/exec-invoke.c

```diff
@@ -2548,6 +2557,96 @@ static int create_many_symlinks(const char *root, const char *source, char **sym
         return 0;
 }
 
+static int apply_exec_quotas(const char *target_dir, const char *cgroup_path, ExecDirectoryType type, QuotaLimit ql) {
+        assert(target_dir);
+        assert(ql.quota_set);
+
+        _cleanup_close_ int fd = -EBADF;
+        int r = 0;
+
+        fd = xopenat(AT_FDCWD, target_dir, O_RDONLY|O_CLOEXEC|O_NOCTTY|O_NOFOLLOW);
+        if (fd < 0)
+                return log_debug_errno(r, "Exec quotas: Failed to open '%s': %m", target_dir);
+
+        /* Generate candidate project id */
+        static const sd_id128_t k = SD_ID128_ARRAY(e1,4a,79,9b,64,40,41,4a,a8,46,c2,f3,f9,19,4f,01);
+        _cleanup_free_ char *proj_id_plain = NULL;
+        r = asprintf(&proj_id_plain, "%s%d", cgroup_path, type);
```

Comment:
the enum numbers are internal, the string representation otoh is supposed to be be stable, hence maybe replace this by:

```c
proj_id_plain = strjoin(cgroup_path, ".", exec_directory_type_to_string(type));
if (!proj_id_plain) 
        …
```